haystack/pipelines/base.py

```diff
@@ -216,6 +280,53 @@ def list_pipelines_on_deepset_cloud(
         pipeline_config_infos = list(client.list_pipeline_configs())
         return pipeline_config_infos
 
+    @classmethod
+    def save_to_deepset_could(
+        cls,
+        query_pipeline: BasePipeline,
+        index_pipeline: BasePipeline,
+        pipeline_config_name: str,
+        workspace: str = "default",
+        api_key: Optional[str] = None,
+        api_endpoint: Optional[str] = None,
+        overwrite: bool = False,
+    ):
+        """
+        Saves a Pipeline config to Deepset Cloud defining the individual components and how they're tied together to form
+        a Pipeline. A single config must declare a query pipeline and a index pipeline.
+
+        :param query_pipeline: the query pipeline to save.
+        :param index_pipeline: the index pipeline to save.
+        :param pipeline_config_name: name of the config file inside the Deepset Cloud workspace.
+        :param workspace: workspace in Deepset Cloud
+        :param api_key: Secret value of the API key.
+                        If not specified, will be read from DEEPSET_CLOUD_API_KEY environment variable.
+        :param api_endpoint: The URL of the Deepset Cloud API.
+                             If not specified, will be read from DEEPSET_CLOUD_API_ENDPOINT environment variable.
+        :param overwrite: Whether to overwrite the config if it already exists. Otherwise an error is being raised.
+        """
+        query_config = query_pipeline.get_config()
+        index_config = index_pipeline.get_config()
+        pipelines = query_config["pipelines"] + index_config["pipelines"]
+        all_components = query_config["components"] + index_config["components"]
+        distinct_components = [c for c in {component["name"]: component for component in all_components}.values()]
+        config = {"components": distinct_components, "pipelines": pipelines, "version": "0.9"}
```

Comment:
Does it maybe make sense to map the pipeline yaml version to the haystack version or add a "supported yaml versions" tag to the haystack package and take the latest one ? 🤔

Reply:
The version and its meaning here is currently more or less undefined. 
@ZanSara, I think this should be part of https://github.com/deepset-ai/haystack/issues/1981, what do you think? 
@ArzelaAscoIi, are there any requirements from DC regarding this?

Reply:
How to deal with YAML versions is still being discussed here: #2042

Reply:
AFAIK, we dont have requirements yet. For us it makes sense to adapt to OSS. :)